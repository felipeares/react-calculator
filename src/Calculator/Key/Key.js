import React from "react";
import classes from "./Key.module.css";
import PropTypes from "prop-types";

const Key = props => (
  <div
    onClick={() => props.clicked(props.value)}
    className={[classes.Key, classes[props.keyType]].join(" ")}
  >
    <p className={classes.KeyValue}>{props.value}</p>
  </div>
);

Key.propTypes = {
  value: PropTypes.string.isRequired,
  clicked: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired
};

export default Key;
