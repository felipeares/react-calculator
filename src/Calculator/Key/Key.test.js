import React from "react";
import { shallow } from "enzyme";
import PropTypes from "prop-types";
import Key from "./Key";

describe("Key", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Key value={""} clicked={jest.fn()} keyType={""} />);
  });

  it("should render main div", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should have a class of Key", () => {
    expect(
      wrapper
        .find("div")
        .first()
        .hasClass("Key")
    ).toEqual(true);
  });

  it("should check for all prop types correctly setup", () => {
    expect(Key.propTypes.value).toBe(PropTypes.string.isRequired);
    expect(Key.propTypes.clicked).toBe(PropTypes.func.isRequired);
    expect(Key.propTypes.keyType).toBe(PropTypes.string.isRequired);
  });

  it("should render as text the value props", () => {
    wrapper.setProps({ value: "test" });
    expect(wrapper.text()).toEqual("test");
  });
});
