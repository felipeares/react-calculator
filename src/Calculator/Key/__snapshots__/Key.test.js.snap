// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key have not changed the UI 1`] = `
<div
  className="Key "
  onClick={[Function]}
>
  <p
    className="KeyValue"
  />
</div>
`;
